src/gordon_gcp/plugins/service/gdns_publisher.py

```diff
@@ -247,15 +247,16 @@ def _format_resource_record_changes(self, action, resource_record):
         """
         record_name = changes['additions'][0]['name']
         record_type = changes['additions'][0]['type']
-
-        response = await self.http_client.get_all(resource_records_url)
+        search_params = {
+            'name': record_name,
+            'type': record_type
+        }
+        response = await self.http_client.get_all(
+            resource_records_url, params=search_params)
         for resp in response:
             for rec in resp['rrsets']:
-                if rec['name'] == record_name and rec['type'] == record_type:
-                    changes['deletions'] = [rec]
-                    break
-
-        return changes
+                changes['deletions'] = [rec]
+                return changes
```

Comment:
I haven't looked closely at the failing tests, but there may be cases (i.e. an empty list in `response`) where you won't ever return `changes` from this method.

Reply:
I would also make this an if, not a loop.  if response, changes = response[0]['rrsets'].